### PR TITLE
[Rebase& FF] 202405: SMM Communicate Buffer Protection

### PR DESCRIPTION
**Pull request by VivianNK**
## Description

Changes are from the following commits:

16f1237fca
0c04a06b24
5c3020b2ae

- [x] Impacts functionality?
- [x] Impacts security?
- [ ] Breaking change?
- [ ] Includes tests?
- [ ] Includes documentation?

## How This Was Tested

Changes from release/202311

## Integration Instructions

N/A

### CODE REVIEW

File: StandaloneMmPkg/Core/StandaloneMmCore.c
Comment (makubacki) [inline StandaloneMmPkg/Core/StandaloneMmCore.c:1]: The "Decouple MM core mailbox data" commits need to be fixed up to be squashed and carry forward the commit author/details from https://github.com/microsoft/mu_basecore/commit/16f1237fca06c8f5b570fc0be1b17e6d0578c2d0

Comment (VivianNK) [reply StandaloneMmPkg/Core/StandaloneMmCore.c:1]: Squashed? I split the one commit into two by package. Should I keep it as one commit? Or just add the author + message from  [16f1237 ](https://github.com/microsoft/mu_basecore/commit/16f1237fca06c8f5b570fc0be1b17e6d0578c2d0) into each of the commits?

Comment (makubacki) [reply StandaloneMmPkg/Core/StandaloneMmCore.c:1]: The split per package is fine, I missed that was the goal there. In that case, just `git commit --amend --no-edit --author="Kun Qin <kuqin@microsoft.com>" --date="Wed Feb 16 20:10:34 2022 +0000"` on each commit so they retain the same info as the original commit.

Comment (makubacki) [reply StandaloneMmPkg/Core/StandaloneMmCore.c:1]: You can drop stale PR info from each commit and adjust the message per the commit contents now.

**Review by kuqin12 — APPROVED** [review]

**Review by os-d — APPROVED** [review]

File: StandaloneMmPkg/Include/Library/StandaloneMmMemLib.h [hunk 1]
```diff
@@ -131,4 +131,20 @@ MmSetMem (
   IN UINT8  Value
   );
```
Comment (makubacki) [inline StandaloneMmPkg/Include/Library/StandaloneMmMemLib.h:133]: Looks like changes in this commit need MU_CHANGE tags.

**Review by makubacki — APPROVED** [review]

**Pull request merged**